compound_disease/compound_treats_disease/edge_prediction_experiment/edge_prediction_experiment.py

```diff
@@ -0,0 +1,330 @@
+
+# coding: utf-8
+
+# # Compound Treats Disease Edge Prediction
+
+# This notebook is designed to take the next step moving from predicted sentences to edge predictions
+
+# In[1]:
+
+
+get_ipython().run_line_magic('load_ext', 'autoreload')
+get_ipython().run_line_magic('autoreload', '2')
+get_ipython().run_line_magic('matplotlib', 'inline')
+
+import pandas as pd
+from sklearn.metrics import precision_recall_curve, roc_curve, auc
+import matplotlib.pyplot as plt
+import seaborn as sns
+
+
+# In[2]:
+
+
+total_candidates_df = (
+    pd
+    .read_table("../dataset_statistics/results/all_ctd_candidates.tsv.xz")
+    .sort_values("candidate_id")
+)
+total_candidates_df.head(2)
+
+
+# In[3]:
+
+
+sentence_prediction_df = (
+    pd
+    .read_table("results/all_predicted_sentences.tsv")
+    .sort_values("candidate_id")
+)
+sentence_prediction_df.head(2)
+
+
+# In[4]:
+
+
+total_candidates_pred_df = (
+    total_candidates_df[[
+    "doid_id", "doid_name", 
+    "drugbank_id", "drug_name", 
+    "text", "hetionet",
+    "candidate_id"
+    ]]
+    .merge(sentence_prediction_df, on="candidate_id")
+)
+total_candidates_pred_df.head(2)
+
+
+# In[5]:
+
+
+grouped_candidates_pred_df=(
+    total_candidates_pred_df
+    .groupby(["doid_id", "drugbank_id"], as_index=False)
+    .agg({
+        "model_prediction": ['max', 'mean', 'median'], 
+        'hetionet': 'max',
+        "drug_name": 'first',
+        "doid_name": 'first'
+    })
+)
+grouped_candidates_pred_df.head(2)
+
+
+# In[6]:
+
+
+grouped_candidates_pred_df.columns = [
+    "_".join(col) 
+    if col[1] != '' and col[0] not in ['hetionet', 'drug_name', 'doid_name'] else col[0] 
+    for col in grouped_candidates_pred_df.columns.values
+]
+grouped_candidates_pred_df.head(2)
+
+
+# In[7]:
+
+
+grouped_candidates_pred_df.to_csv(
+    "results/combined_predicted_ctd_sentences.tsv.xz", 
+    index=False, sep="\t", compression="xz"
+)
+
+
+# In[8]:
+
+
+grouped_candidates_pred_df.hetionet.value_counts()
+
+
+# In[9]:
+
+
+performance_map = {}
+
+
+# In[10]:
+
+
+precision, recall, pr_threshold = precision_recall_curve(
```

Comment:
So this curve is showing the performance of the prediction using the max, mean, median of what score exactly?

Reply:
So each mention pair has a change of sharing a particular relationship. After my classification pipeline, each sentence that contains the given pair has a confidence score for establishing existence. A pair can have multiple sentences, so combining each score is unintuitive (i.e. is taking the max score appropriate here? how about median or mean?). The plot here shows the performance for doing each operation (max, mean, median) separately. Based on the results Max score seems the most appropriate.